src/Microsoft.Extensions.Logging.Testing/Xunit/LoggedTestInvoker.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+using Xunit.Abstractions;
+using Xunit.Sdk;
+
+namespace Microsoft.Extensions.Logging.Testing
+{
+    public class LoggedTestInvoker : XunitTestInvoker
+    {
+        public LoggedTestInvoker(ITest test, IMessageBus messageBus, Type testClass, object[] constructorArguments, MethodInfo testMethod, object[] testMethodArguments, IReadOnlyList<BeforeAfterTestAttribute> beforeAfterAttributes, ExceptionAggregator aggregator, CancellationTokenSource cancellationTokenSource) : base(test, messageBus, testClass, constructorArguments, testMethod, testMethodArguments, beforeAfterAttributes, aggregator, cancellationTokenSource)
+        {
+        }
+
+        protected override object CreateTestClass()
+        {
+            var testClass = base.CreateTestClass();
+
+            if (testClass is LoggedTest loggedTestClass)
```

Comment:
Can't we do it for any class, so you're not forced to inherit and can just inject the constructor argument?

Reply:
We are not injecting it as a constructor argument. We are setting the property of the base class LoggedTest.

Reply:
Unless you are suggesting we change the design?

Reply:
I always thought we are going argument injection, that's why we started with argument injection DI prototype.

Reply:
Found a way of removing the need to flow the ITestOutputHelper with constructors.